support angular 9

webcat12345 opened the pull request:
Current `@ngui` projects are not working on new Angular 9.

#### Completed:

* migrate to `@angular/cli` project
* support Angular 9

#### TODO:

* ~directive support~
* ~update relevant libraries for demo - `@ngui/utils`~ (removed @ngui/utils usage from demo)
* ~demo implementation~
* ~update documents and change log and new version~

~**Important: The status of this PR is WIP (work in progress), just confirming that you are willing to accept this kind of update. Please let me know if you guys are going to approve this PR, of course after complete**~

webcat12345 commented:
@almothafar This pull request is ready to be merged. Please review it. Thank you.

almothafar commented on package.json:
```diff
@@ -1,78 +1,52 @@
-  "main": "dist/auto-complete.umd.js",
-  "module": "dist/index.js",
-  "typings": "dist/index.d.ts",
+  "name": "auto-complete",
```
```suggestion
  "name": "@ngui/auto-complete",
```
Package name should be still `@ngui/auto-complete`

webcat12345 replied:
It should be fixed. But refer - `auto-complate/projects/auto-complete/package.json` file.

almothafar commented on package.json:
```diff
@@ -1,78 +1,52 @@
-  "module": "dist/index.js",
-  "typings": "dist/index.d.ts",
+  "name": "auto-complete",
+  "version": "0.0.0",
```
```suggestion
  "version": "3.0.0",
```
Because it got a lot of breaking changes please make it version 3.0.0

webcat12345 replied:
It should be fixed. But refer - `auto-complate/projects/auto-complete/package.json` file.

almothafar commented on package.json:
```diff
@@ -1,78 +1,52 @@
+    "@angular/router": "~9.0.0",
+    "rxjs": "~6.5.4",
+    "tslib": "^1.10.0",
+    "zone.js": "~0.10.2"
```
There are a lot of dependencies, you need to move them to dev if you want them only for dev, anything optional "and not a must" should be peer, like material, dependencies here for what required to get this works.

webcat12345 replied:
`/auto-complete/projects/auto-complete/package.json`

Please check this `package.json`. Peer dependencies are listed there. The current file is for the base.

webcat12345 replied:
It should be fixed. But refer - `auto-complate/projects/auto-complete/package.json` file.

almothafar commented on package.json:
```diff
@@ -1,78 +1,52 @@
-    "auto-complete",
-    "input",
-    "select"
-  ]
```
Please keep these.

webcat12345 replied:
It should be fixed. But refer - `auto-complate/projects/auto-complete/package.json` file.

almothafar commented on package.json:
```diff
@@ -1,78 +1,52 @@
-  "name": "@ngui/auto-complete",
-  "version": "2.0.0",
-  "description": "Angular Input Autocomplete",
-  "license": "MIT",
```
Why these lines removed?

webcat12345 replied:
`/auto-complete/projects/auto-complete/package.json`

Library `package.json` file should be here. This `package.json` is for the base project.

webcat12345 replied:
It should be fixed. But refer - `auto-complate/projects/auto-complete/package.json` file.

webcat12345 commented:
Thanks for reviewing @almothafar .
I have updated npm script - `build-lib:prod`, please check the last commit.
FYI, `auto-complete/package.json` is just the base project which is mainly running the demo. Library package.json should be there in the `auto-complete/projects/auto-complete/package.json`.
I have updated it according to your comments, you can verify if those are correctly formed.

webcat12345 commented:
@almothafar Could you please review this PR soon? Actually our product is depending on this library and we can't update to the Angular 9 due to this problem. Thank you.

almothafar commented:
@webcat12345 sorry for the delay, as I'm moving to a new house and my internet is not set up yet, I merged it, but still, I need to test and distribute it.

almothafar commented:
Just to keep you updated, still without the internet until now, hopefully, this week ISP set up it.

tpapagiannis commented:
Still without ISP ?

almothafar commented:
@tpapagiannis it is merged and released already under version 3.0.0, if you have a problem, please say it straight.
